src/MongoDB.Driver.Core/Core/Compression/Zstd/ZstandardStream.cs

```
@@ -0,0 +1,382 @@
+                    break;
+                case CompressionMode.Decompress:
+                    _streamReadHelper = new StreamReadHelper(
+                        compressedStream: compressedStream,
+                        compressedBuffer: _arrayPool.Rent(_nativeWrapper.RecommendedInputSize));
+                    break;
+            }
+        }
+
+        public override bool CanRead => _compressedStream.CanRead && _compressionMode == CompressionMode.Decompress;
+
+        public override bool CanWrite => _compressedStream.CanWrite && _compressionMode == CompressionMode.Compress;
+
+        public override bool CanSeek => false;
+
+        public override long Length => throw new NotSupportedException();
+
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+
+            if (!_disposed)
+            {
+                switch (_compressionMode)
+                {
+                    case CompressionMode.Compress:
+                        try
+                        {
+                            if (!_flushed)
+                            {
+                                try
+                                {
+                                    Flush();
+                                }
+                                catch
+                                {
+                                    // ignore exception
+                                }
+                            }
+                        }
+                        finally
+                        {
+                            _nativeWrapper.Dispose();
+                            _arrayPool.Return(_streamWriteHelper.CompressedBufferInfo.Bytes);
+                        }
```

I thought we had discussed that finally is not appropriate here. Finally is typically used to do cleanup when there is a failure, not as a way to ignore exceptions. Lines 119-137 should be:

    if (!_flushed)
    {
        try { Flush(); } catch { }
    }
    _nativeWrapper.Dispose();
    _arrayPool.Return(_streamWriteHelper.CompressedBufferInfo.Bytes);

---

Done